src/scry/publish_diagnostic.cr

```diff
@@ -6,20 +6,31 @@ require "./workspace"
 
 module Scry
   struct PublishDiagnostic
-    METHOD = "textDocument/publishDiagnostics"
+    FILES_WITH_DIAGNOSTIC = [] of String
```

Comment:
huh I was talking about this constant :joy:

Reply:
Well, this constant is required, `FILES_WITH_DIAGNOSTIC` is updated only during `@diagnostic.from`, so, if I use an ivar, then next time I fix my code, I won't be able to clean the past errors :sweat_smile: 

I could use a class var as well, but I think a constant is cleaner

Reply:
Oh ok, could you rename it as `FILES_WITH_DIAGNOSTIC` with a little comment about that?

Reply:
@bew Done!, I just discovered this was clearing all diagnostics, even those not generated by a current file, so, I'm using a Hash now to group diagnostic per file:

![Imgur](https://i.imgur.com/pgAcUq6.gif)